api/services/tools/api_tools_manage_service.py

```diff
@@ -113,6 +113,8 @@ def create_api_tool_provider(
         if schema_type not in [member.value for member in ApiProviderSchemaType]:
             raise ValueError(f"invalid schema type {schema}")
 
+        provider_name = provider_name.strip()
```

Comment:
Could you explain why this line is necessary?

Reply:
Additionally, if the provider_name starts with spaces, these spaces can be correctly encoded and transmitted, but they will not be rendered in the tool list. 
Since Dify does not allow duplicate provider names, it seems a bit pointless to permit leading and trailing spaces.